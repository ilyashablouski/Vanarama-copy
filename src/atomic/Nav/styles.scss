$color: black;

nav {
  background-color: black;
  width: 100%;
  a {
    color: #fff;
    @include textShadow;
  }
}

.nav-elements {
  display: inline;
}
