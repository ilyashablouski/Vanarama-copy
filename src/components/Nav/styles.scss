nav {
  a {
    color: $color;
    @include textShadow;
  }
}
