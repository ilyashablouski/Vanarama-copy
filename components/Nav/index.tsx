import React from "react"
import Link from "next/link"

import "./styles.scss"

const Nav = () => (
  <nav>
    <ul>
      <li>
        <Link prefetch href="/">
          <a>Home</a>
        </Link>
      </li>
      <li>
        <Link prefetch href="/test/">
          <a>Test</a>
        </Link>
      </li>
      <li>
        <Link prefetch href="/sub/test/">
          <a>Sub Test</a>
        </Link>
      </li>
    </ul>
  </nav>
)

export default Nav
