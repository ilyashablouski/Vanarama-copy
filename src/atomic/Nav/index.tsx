import React, { useEffect } from "react"
import Link from "next/link"
import { useRouter } from "next/router"

import "./styles.scss"

import Phone from "atomic/atoms/Phone"

const Nav = () => {
  const router = useRouter()
  useEffect(() => {
    router.prefetch("/test")
  })
  return (
    <nav className="nav-elements">
      <Phone />
      <Link href="/">
        <a>Home</a>
      </Link>
      <Link href="/test">
        <a>Test</a>
      </Link>
      <Link href="/sub/test">
        <a>Sub Test</a>
      </Link>
    </nav>
  )
}

export default Nav
