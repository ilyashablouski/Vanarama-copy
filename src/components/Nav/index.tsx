import React, { useEffect } from 'react';
import Link from 'next/link';
import { useRouter } from 'next/router';

import './styles.scss';

const Nav = () => {
  const router = useRouter();
  useEffect(() => {
    router.prefetch('/test');
  });
  return (
    <nav className="nav-elements">
      <Link href="tel:+01442 507665">
        <a>01442 507665</a>
      </Link>
      <Link href="/">
        <a>Home</a>
      </Link>
      <Link href="/test">
        <a>Test</a>
      </Link>
      <Link href="/sub/test">
        <a>Sub Test</a>
      </Link>
    </nav>
  );
};

export default Nav;
