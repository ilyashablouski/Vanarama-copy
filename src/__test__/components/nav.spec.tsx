import Nav from "../../components/Nav"
import * as React from "react"
import { shallow } from "enzyme"

describe("<Nav /> rendering", () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Nav />)
  })

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot()
  })
})
