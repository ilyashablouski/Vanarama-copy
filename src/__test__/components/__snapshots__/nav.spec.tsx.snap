// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> rendering renders correctly 1`] = `
<nav
  className="nav-elements"
>
  <Link
    href="tel:+01442 507665"
  >
    <a>
      01442 507665
    </a>
  </Link>
  <Link
    href="/"
  >
    <a>
      Home
    </a>
  </Link>
  <Link
    href="/test"
  >
    <a>
      Test
    </a>
  </Link>
  <Link
    href="/sub/test"
  >
    <a>
      Sub Test
    </a>
  </Link>
</nav>
`;
